traffic_ops/app/lib/API/Deliveryservice.pm

```diff
@@ -442,6 +442,139 @@ sub update {
 	}
 }
 
+sub safe_update {
+	my $self   = shift;
+	my $id     = $self->param('id');
+	my $params = $self->req->json;
+
+
+	my $helper = new Utils::Helper( { mojo => $self } );
+
+	my $ds = $self->db->resultset('Deliveryservice')->find( { id => $id } );
+	if ( !defined($ds) ) {
+		return $self->not_found();
+	}
+
+
+	if ( &is_portal($self) && $helper->is_delivery_service_assigned($id) ) {
+
+		my $xml_id = $params->{xmlId};
+		if ( $ds->xml_id ne $xml_id ) {
+			my $existing = $self->db->resultset('Deliveryservice')->find( { xml_id => $xml_id } );
+			if ($existing) {
+				return $self->alert( "A deliveryservice with xmlId " . $xml_id . " already exists." );
+			}
+		}
+
+		my $values = {
+			display_name           => $params->{displayName},
+			info_url               => $params->{infoUrl},
+			long_desc              => $params->{longDesc},
+			long_desc_1            => $params->{longDesc1},
+			long_desc_2            => $params->{longDesc2},
+		};
+
+		my $rs = $ds->update($values);
+		if ($rs) {
+
+			# create location parameters for header_rewrite*, regex_remap* and cacheurl* config files if necessary
```

Comment:
I don't think you need to do lines 481-484, actually you probably shouldn't be because this would imply that i can pass in a new profileId or xmlId and make significant changes to the delivery service when really they can change xmlId and profileId all day long and it "should" have no effect in this route.